sos/component.py

```diff
@@ -384,14 +388,40 @@ def _setup_logging(self):
             ui_fhandler.setFormatter(logging.Formatter(
                 '%(asctime)s %(levelname)s: %(message)s'))
 
-            self.ui_log.addHandler(ui_fhandler)
+            ui_fhandler.name = 'ui_fhandler'
+            self.add_handler(self.ui_log, ui_fhandler)
 
         if not self.opts.quiet:
             self.add_ui_log_to_stdout()
 
     def get_temp_file(self):
         return self.tempfile_util.new()
 
+    def handler_present(self, name):
+        result = False
+        loggers = []
+        if hasattr(self, 'soslog'):
+            loggers.append(self.soslog)
+        if hasattr(self, 'ui_log'):
+            loggers.append(self.ui_log)
+        for logger in loggers:
+            for handler in logger.handlers:
+                if handler.name == name:
+                    result = True
+                    break
+        return result
```

Comment:
As we are iterating over 2 loggers / names, isn't it simplier to call

    if name in ['soslog', 'ui_log']:
        return hasattr(self, name)
    return False

or even simplier:

    return name in ['soslog', 'ui_log'] and hasattr(self, name)

?

Reply:
above will return string instance, but we need logging.Handler instance, but i can shorten it a bit